src/main/java/org/terasology/rendering/nui/layers/hud/DirectionalDamageOverlay.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.rendering.nui.layers.hud;
+
+import org.terasology.engine.Time;
+import org.terasology.math.Direction;
+import org.terasology.nui.Canvas;
+import org.terasology.nui.widgets.UIImage;
+import org.terasology.registry.In;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class DirectionalDamageOverlay extends CoreHudWidget {
+
+    @In
+    private Time time;
+
+    private UIImage bottom;
+    private UIImage top;
+    private UIImage left;
+    private UIImage right;
+
+    private Map<Direction, TimingInformation> activeIndicators = new HashMap<>();
+
+    @Override
+    public void initialise() {
+        bottom = find("damageBottom", UIImage.class);
+        top = find("damageTop", UIImage.class);
+        left = find("damageLeft", UIImage.class);
+        right = find("damageRight", UIImage.class);
+    }
+
+    private Optional<UIImage> imageForDirection(Direction direction) {
+        switch (direction) {
+
+            case FORWARD:
+                return Optional.of(top);
+            case RIGHT:
+                return Optional.of(right);
+            case LEFT:
+                return Optional.of(left);
+            case BACKWARD:
+                return Optional.of(bottom);
+            case UP:
+            case DOWN:
```

Comment:
Should these cases return top / bottom? :thinking:

Reply:
I couldn't wrap my head around how to best render the indicators if one indicator is used for multiple directions... =/ I think the data structures as they are don't fit that use case.

The concrete problem I had (why the bottom indicator did not show up) was that for `FORWARD`/`UP` the order was "correct", and I first set the image visibility to `false` and then to `true`. For `BACKWARD`/`DOWN` it was the other way around, the second direction canceled out the visibility..

Maybe it would be better to iterate over the images and do the lookup the other way around, and then add some _one-of_ logic to determine whether the indicator is visible...

Reply:
Can you write up a follow-up issue for that? Same for the damage indicator for falling and thirst/hunger damage.

Reply:
I'd rather directly do the follow-up PR...